x-pack/plugins/fleet/server/services/epm/elasticsearch/template/template.ts

```diff
@@ -1014,8 +1014,13 @@ const updateExistingDataStream = async ({
     skipDataStreamRollover?: boolean;
   };
 }) => {
+  const dataStream = await esClient.indices.getDataStream({
+    name: dataStreamName,
+    expand_wildcards: ['open', 'hidden'],
+  });
+
   const existingDs = await esClient.indices.get({
-    index: dataStreamName,
+    index: dataStream?.data_streams?.[0]?.indices?.at(-1)?.index_name ?? dataStreamName,
```

Comment:
what is the difference between using get data stream vs get indices?

could we pass in `features` and `include_defaults` [query params](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-get-index.html#get-index-api-query-params) to minify the response from this ES API instead?

Reply:
We still need to do a call to get indices to get the current settings but only for the current write index, not all the stream backing indices.
I found that we already call get datastream before in the code, so with a little refactor I was able to remove that code let me know what do you think of it.